pyDmdReader/_dmd_reader.py

```diff
@@ -170,19 +172,19 @@ def read_dataframe(
         return full_frame
 
     def read_array(
-        self,
-        ch_names: Union[List[str], str, List[int], int],
-        start_time: float = 0.0,
-        end_time: Optional[float] = None,
-        timestamp_format: TimestampFormat = TimestampFormat.SECONDS_SINCE_START,
-        max_samples: Optional[int] = None
+            self,
```

Comment:
Since `read_array` is a public method, we should not change its parameters/names so we do not break anything for existing code. It would be better to add a new method `read_array_by_index` or similar.
The index-version should has the parameters `start_sample` and `end_sample` (exclusive range?) with only accept ints.

Reply:
done